packages/syft/tests/syft/core/tensor/passthrough_test.py

```diff
@@ -1,12 +1,970 @@
+    assert result_b == tensor_c
+
+
+def test__rrshift__() -> None:
+    data_a = np.array([1, 2, -1], dtype=np.int32)
+    data_b = torch.tensor([0, 2, 1], dtype=torch.int32)
+    expected = torch.tensor([0, 8, -1], dtype=torch.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__rrshift__(tensor_b)
+    result_b = tensor_a.__rrshift__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__pow__() -> None:
+    data_a = np.array([1, 2, -1], dtype=np.int32)
+    data_b = np.array([0, 2, 1], dtype=np.int32)
+    expected = np.array([1, 4, -1], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__pow__(tensor_b)
+    result_b = tensor_a.__pow__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__rpow__() -> None:
+    data_a = np.array([1, 2, 1], dtype=np.int32)
+    data_b = torch.tensor([0, 2, -1], dtype=torch.int32)
+    expected = torch.tensor([0, 4, -1], dtype=torch.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__rpow__(tensor_b)
+    result_b = tensor_a.__rpow__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__divmod__() -> None:
+    data_a = np.array([1, 2, 3], dtype=np.int32)
+    data_b = np.array([1, 5, 4], dtype=np.int32)
+    expected = np.array([0, 1, 3], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__divmod__(tensor_b)
+    result_b = tensor_a.__divmod__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__neg__() -> None:
+    data_a = np.array([1, 0, -1], dtype=np.int32)
+    expected = np.array([-1, 0, 1], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=expected)
+
+    assert tensor_a.__neg__() == tensor_b
+
+
+def test__invert__() -> None:
+    data_a = np.array([-1, 0, 1], dtype=np.int32)
+    expected = np.array([0, -1, -2], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=tensor_a.__invert__())
+    tensor_c = PassthroughTensor(child=expected)
+
+    assert tensor_b == tensor_c
+
+
+def test__index__() -> None:
+    data_a = np.array([1, 2, 3], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+
+    assert tensor_a[2].__index__() == 3
+
+
+def test_copy() -> None:
+    data_a = np.array([1, 2, 3], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+
+    assert tensor_a.copy() == tensor_a
```

Comment:
This should also assert that the copy is not the same object in memory with something like:
```
assert id(tensor_a_copy) != id(tensor_a)
```

Reply:
Added suggestion, changed to:    assert id(tensor_a.copy()) != id(tensor_a)